From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Sun, 15 Nov 2020 17:08:15 +0100
Subject: smarcard: Add utility function to get the login token name

We repeat the same call with the env variable we watch multiple times,
being a bit prone to errors, let's move this to a function call to avoid
issues.

Origin: https://gitlab.gnome.org/GNOME/gnome-settings-daemon/-/merge_requests/208
---
 plugins/smartcard/gsd-smartcard-manager.c | 4 ++--
 plugins/smartcard/gsd-smartcard-service.c | 4 ++--
 plugins/smartcard/gsd-smartcard-utils.c   | 6 ++++++
 plugins/smartcard/gsd-smartcard-utils.h   | 2 ++
 4 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/plugins/smartcard/gsd-smartcard-manager.c b/plugins/smartcard/gsd-smartcard-manager.c
index b7b768f..9e49195 100644
--- a/plugins/smartcard/gsd-smartcard-manager.c
+++ b/plugins/smartcard/gsd-smartcard-manager.c
@@ -564,7 +564,7 @@ on_modules_initialized (GObject      *source_object,
 
         login_token = gsd_smartcard_manager_get_login_token (self);
 
-        if (login_token || g_getenv ("PKCS11_LOGIN_TOKEN_NAME") != NULL) {
+        if (login_token || gsd_smartcard_utils_get_login_token_name () != NULL) {
                 if (!login_token ||
                     !gsd_smartcard_utils_slot_has_flags (login_token, CKF_TOKEN_PRESENT))
                         gsd_smartcard_manager_do_remove_action (self);
@@ -763,7 +763,7 @@ get_login_token_for_operation (GsdSmartcardManager      *self,
                 GckTokenInfo *token_info = value;
                 const char *token_name = token_info->label;
 
-                if (g_strcmp0 (g_getenv ("PKCS11_LOGIN_TOKEN_NAME"), token_name) == 0)
+                if (g_strcmp0 (gsd_smartcard_utils_get_login_token_name (), token_name) == 0)
                         return g_object_ref (card_slot);
         }
 
diff --git a/plugins/smartcard/gsd-smartcard-service.c b/plugins/smartcard/gsd-smartcard-service.c
index 8fffbaa..2a7abec 100644
--- a/plugins/smartcard/gsd-smartcard-service.c
+++ b/plugins/smartcard/gsd-smartcard-service.c
@@ -125,7 +125,7 @@ register_login_token_alias (GsdSmartcardService *self)
         const char *object_path;
         const char *token_name;
 
-        token_name = g_getenv ("PKCS11_LOGIN_TOKEN_NAME");
+        token_name = gsd_smartcard_utils_get_login_token_name ();
 
         if (token_name == NULL)
                 return;
@@ -559,7 +559,7 @@ synchronize_token_now (GsdSmartcardService *self,
                         token_name = token_info->label;
         }
 
-        if (g_strcmp0 (g_getenv ("PKCS11_LOGIN_TOKEN_NAME"), token_name) == 0)
+        if (g_strcmp0 (gsd_smartcard_utils_get_login_token_name (), token_name) == 0)
                 is_login_card = is_present;
         else
                 is_login_card = FALSE;
diff --git a/plugins/smartcard/gsd-smartcard-utils.c b/plugins/smartcard/gsd-smartcard-utils.c
index 1f06e56..2dfc17c 100644
--- a/plugins/smartcard/gsd-smartcard-utils.c
+++ b/plugins/smartcard/gsd-smartcard-utils.c
@@ -169,6 +169,12 @@ gsd_smartcard_utils_escape_object_path (const char *unescaped_string)
   return object_path;
 }
 
+const char *
+gsd_smartcard_utils_get_login_token_name (void)
+{
+        return g_getenv ("PKCS11_LOGIN_TOKEN_NAME");
+}
+
 gboolean
 gsd_smartcard_utils_slot_has_flags (GckSlot *slot,
                                     gulong   flags)
diff --git a/plugins/smartcard/gsd-smartcard-utils.h b/plugins/smartcard/gsd-smartcard-utils.h
index 0e6c5b6..b6f9af9 100644
--- a/plugins/smartcard/gsd-smartcard-utils.h
+++ b/plugins/smartcard/gsd-smartcard-utils.h
@@ -29,6 +29,8 @@ void             gsd_smartcard_utils_register_error_domain             (GQuark e
                                                                         GType error_enum);
 char *           gsd_smartcard_utils_escape_object_path                (const char *unescaped_string);
 
+const char *     gsd_smartcard_utils_get_login_token_name              (void);
+
 gboolean         gsd_smartcard_utils_slot_has_flags                    (GckSlot *, gulong flags);
 
 G_END_DECLS
